CMakeLists.txt

```diff
@@ -155,6 +155,13 @@ if (ENABLE_USB_CAMERA)
     createExtension(USB-CAMERA-EXTENSIONS "USB CAMERA EXTENSIONS" "This enables USB camera support" "extensions/usb-camera" "${TEST_DIR}/usb-camera-tests" "TRUE" "thirdparty/libuvc-0.0.6")
 endif()
 
+## TensorFlow extensions
+## Disabled by default because TF can be complex/environment-specific to build
+option(ENABLE_TENSORFLOW "Disables the TensorFlow extensions." OFF)
+if (ENABLE_TENSORFLOW)
+    createExtension(TENSORFLOW-EXTENSIONS "TENSORFLOW EXTENSIONS" "This enables TensorFlow support" "extensions/tensorflow" "${TEST_DIR}/tensorflow-tests")
```

Comment:
does TF provide source built as third party library. in case we want to build for other target like ARM, etc

Reply:
This extension should be compatible with that platform. The key thing would be building TF on (or for, with cross-compiling) the target.

We are dynamically linking TF so it just has to be available in the target linking environment.

As for build time, our custom FindTensorFlow.cmake needs to find the libtensorflow_cc.so and headers.

Ultimately, it would be nice to have the ability to build TF in thirdparty/ and link statically, but there's currently no way to static link TF without giving up important features such as GPU acceleration.

My thought was to keep it as an externally-provided and dynamically-linked library, and keep an eye on libtensorflow_cc as it matures and hopefully supports static linking options and cleaner embedded cmake files we could put in thirdparty/.

Reply:
@phrocker also look at BUILDING.md if you haven't looked at it yet. While there are only instructions for CentOS 7 currently, the procedure should be similar for other environments (installing TF-deps like CUDA/cuDNN, building & installing TF to /usr/local).